model/adb_auth.go

```diff
@@ -263,3 +389,55 @@ func RemoveUser(db *sqlx.DB, userID int) (int, error) {
 
 	return userID, nil
 }
+
+func CreateUserRole(db *sqlx.DB, userRole UserRole) (int, error) {
+	if userRole.UserID == 0 {
+		return 0, errors.New("Invalid User ID")
+	}
+
+	if userRole.Role == "" {
+		return userRole.UserID, errors.New("Role cannot be empty")
+	}
+
+	_, err := db.Exec(`
+INSERT INTO users_roles (user_id, role)
+VALUES (?, ?)
+`, userRole.UserID, userRole.Role)
+
+	if err != nil {
+		return userRole.UserID, errors.Wrapf(err, "Could not add User Role for User %d", userRole.UserID)
+	}
+
+	return userRole.UserID, nil
+}
+
+func RemoveUserRole(db *sqlx.DB, userRole UserRole) (int, error) {
+	if userRole.UserID == 0 {
+		return 0, errors.New("Invalid User ID")
+	}
+
+	tx, err := db.Beginx()
```

Comment:
I don't think you need a transaction for just one SQL statement?

Reply:
Probably not.. but it would make it easier if this method got more complex later on and needed to do more than just one query/operation.